zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/FastLeaderElection.java

```diff
@@ -976,26 +976,9 @@ public Vote lookForLeader() throws InterruptedException {
                     /*
                      * Exponential backoff
                      */
-                    int tmpTimeOut = notTimeout * 2;
-                    notTimeout = Math.min(tmpTimeOut, maxNotificationInterval);
-
-                    /*
-                     * When a leader failure happens on a master, the backup will be supposed to receive the honour from
-                     * Oracle and become a leader, but the honour is likely to be delay. We do a re-check once timeout happens
-                     *
-                     * The leader election algorithm does not provide the ability of electing a leader from a single instance
-                     * which is in a configuration of 2 instances.
-                     * */
-                    self.getQuorumVerifier().revalidateVoteset(voteSet, notTimeout != minNotificationInterval);
-                    if (self.getQuorumVerifier() instanceof QuorumOracleMaj && voteSet != null && voteSet.hasAllQuorums() && notTimeout != minNotificationInterval) {
```

Comment:
`voteSet != null && voteSet.hasAllQuorums() && notTimeout != minNotificationInterval` 
actually is as same as
`self.getQuorumVerifier().revalidateVoteset(voteSet, notTimeout != minNotificationInterval);`

what @maoling suggested is to remove line 989, and replace the second part of the condition with the function.

The If-statement on line 990 shall remain.

Reply:
all right, i got it wrong, i will rollback it later

Reply:
thanks! I appreciate your effort :). That was my miss when I introduced these codes a while ago.

Reply:
thanks for your reminder